dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/template/components/IntegerComponentFactory.kt

```diff
@@ -0,0 +1,43 @@
+package org.dataland.frameworktoolbox.template.components
+
+import org.dataland.frameworktoolbox.intermediate.components.ComponentBase
+import org.dataland.frameworktoolbox.intermediate.components.IntegerComponent
+import org.dataland.frameworktoolbox.intermediate.group.ComponentGroupApi
+import org.dataland.frameworktoolbox.intermediate.group.create
+import org.dataland.frameworktoolbox.template.TemplateDiagnostic
+import org.dataland.frameworktoolbox.template.model.TemplateRow
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Generates IntegerComponents from rows with the component "Integer"
+ */
+@Component
+class IntegerComponentFactory(@Autowired val templateDiagnostic: TemplateDiagnostic) : TemplateComponentFactory {
+    override fun canGenerateComponent(row: TemplateRow): Boolean = row.component == "Integer"
+
+    override fun generateComponent(
+        row: TemplateRow,
+        utils: ComponentGenerationUtils,
+        componentGroup: ComponentGroupApi,
+    ): ComponentBase {
+        templateDiagnostic.optionsNotUsed(row)
```

Comment:
Again, why do we not use units for decimals in the corresponding factory? Just because there is no use case?

Reply:
We probably did not notice because there simply is no decimal field with a unit attached to it in the excel of Gdv.

Reply:
Then we will added when the time has come.